churaverse-plugins-server/src/churarenPlugin/churarenBossAttackPlugin/domain/bossAttackService.ts

```diff
@@ -0,0 +1,77 @@
+import { IBossAttackRepository } from './IBossAttackRepository'
+import { BossAttack } from './bossAttack'
+import { WorldMap } from '@churaverse/map-plugin-server/domain/worldMap'
+import { IMessageSender } from '@churaverse/network-plugin-server/interface/IMessageSender'
+import { IMainScene, Direction, IEventBus, Position, EntitySpawnEvent } from 'churaverse-engine-server'
+import { BossAttackSpawnMessage, BossAttackSpawnData } from '../message/bossAttackSpawnMessage'
+import { uniqueId } from '@churaverse/churaren-core-plugin-server/utils/uniqueId'
+
+/**
+ * 衝突した or 消滅時間に達したボス攻撃を削除
+ * @param onDelete 削除時に実行する関数.引数に削除されるボス攻撃のインスタンスとidを取る
+ */
+export function removeDieBossAttack(
+  bossAttacks: IBossAttackRepository,
+  onDelete: (bossAttackId: string) => void
+): void {
+  bossAttacks.getAllId().forEach((bossAttackId) => {
+    const bossAttack = bossAttacks.get(bossAttackId)
+    if ((bossAttack?.isDead ?? false) === true) {
```

Comment:
null合体があるのでややこしいですが、ifは基本以下のように書くのがシンプルです
```ts
    if (bossAttack.isDead) {
```

以下だと冗長ですね
```ts
    if (bossAttack.isDead === true) {
```

null合体込みだとしても、boolのリテラルが複数出てくるのは混乱しやすいので以下のような形が見やすいかなと思います
```suggestion
    if (bossAttack?.isDead ?? false) {
```

---

あるいは、型システムが正確に動いてくれそうなのであればアンラップしてあげるのがわかりやすいですね
```ts
const bossAttack = bossAttacks.get(bossAttackId)
if (bossAttack === undefined) return
if (bossAttack.isDead) {
```